packages/core/core/src/RequestTracker.js

```diff
@@ -1543,55 +1548,84 @@ async function loadRequestGraph(options): Async<RequestGraph> {
-    );
+    try {
+      let {requestGraph} = await readAndDeserializeRequestGraph(
+        options.cache,
+        requestGraphKey,
+        cacheKey,
+      );
 
-    let opts = getWatcherOptions(options);
-    let snapshotKey = `snapshot-${cacheKey}`;
-    let snapshotPath = path.join(options.cacheDir, snapshotKey + '.txt');
+      let opts = getWatcherOptions(options);
+
+      timeout = setTimeout(() => {
+        logger.warn({
+          origin: '@parcel/core',
+          message: `Retrieving file system events since last build...\nThis can take upto a minute after branch changes or npm/yarn installs.`,
+        });
+      }, 5000);
+      let startTime = Date.now();
+      let events = await options.inputFS.getEventsSince(
+        options.watchDir,
+        snapshotPath,
+        opts,
+      );
+      clearTimeout(timeout);
 
-    let timeout = setTimeout(() => {
-      logger.warn({
+      logger.verbose({
         origin: '@parcel/core',
-        message: `Retrieving file system events since last build...\nThis can take upto a minute after branch changes or npm/yarn installs.`,
+        message: `File system event count: ${events.length}`,
+        meta: {
+          trackableEvent: 'watcher_events_count',
+          watcherEventCount: events.length,
+          duration: Date.now() - startTime,
+        },
       });
-    }, 5000);
-    let startTime = Date.now();
-    let events = await options.inputFS.getEventsSince(
-      options.watchDir,
-      snapshotPath,
-      opts,
-    );
-    clearTimeout(timeout);
-
-    logger.verbose({
-      origin: '@parcel/core',
-      message: `File system event count: ${events.length}`,
-      meta: {
-        trackableEvent: 'watcher_events_count',
-        watcherEventCount: events.length,
-        duration: Date.now() - startTime,
-      },
-    });
 
-    requestGraph.invalidateUnpredictableNodes();
-    requestGraph.invalidateOnBuildNodes();
-    requestGraph.invalidateEnvNodes(options.env);
-    requestGraph.invalidateOptionNodes(options);
+      requestGraph.invalidateUnpredictableNodes();
+      requestGraph.invalidateOnBuildNodes();
+      requestGraph.invalidateEnvNodes(options.env);
+      requestGraph.invalidateOptionNodes(options);
 
-    try {
       await requestGraph.respondToFSEvents(
         options.unstableFileInvalidations || events,
         options,
         10000,
       );
       return requestGraph;
     } catch (e) {
+      if (timeout != null) {
+        clearTimeout(timeout);
+      }
+      let additionalMessage;
+      let logEvent;
+      if (e.message && e.message.includes('invalid clockspec')) {
+        logEvent = {
+          origin: '@parcel/core',
+          message: `Error reading clockspec from snapshot, building with clean cache.`,
+          meta: {
+            additionalMessage: ``,
+            trackableEvent: 'invalid_clockspec_error',
+          },
+        };
+        additionalMessage = readFileSync(snapshotPath).toString('utf-8');
```

Comment:
You shouldn't used `readFileSync` from `fs` directly, you should use `options.inputFS.readFileSync`